src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -233,6 +237,29 @@ export class NotebookModel extends Disposable implements INotebookModel {
 
 	public set trustedMode(isTrusted: boolean) {
 		this._trustedMode = isTrusted;
+		let whitelistedFolders: string[] = this.workspaceContextService.getValue('notebook.trustedNotebooks');
+		let workspace: Workspace = this.workspaceContextService.getWorkspace();
+		let workspacePaths: string[] = workspace!.folders!.map(uri => uri.uri.fsPath);
+		let modifiedWhitelistedFolder: boolean = false;
+		workspacePaths.forEach(wsPath => {
+			// tslint:disable-next-line: triple-equals
+			let wsIndex = whitelistedFolders.findIndex(trustedFolder => trustedFolder === wsPath);
+
+			if (isTrusted && wsIndex === -1) {
+				whitelistedFolders.push(wsPath);
```

Comment:
Since you're trusting a directory that means that if a new notebook is added later on it'll automatically be trusted which is not something that's clear to me given the description of the feature. I would rather see this split into two separate features - one that allows you to trust all notebooks in a folder/sub-folders (but only a one-time thing, new notebooks added still require explicit trusting) and another that lets you whitelist folders for trusting. Those are separate enough use cases and useful enough in their own rights to warrant having both IMO